client/asset/dcr/dcr.go

```diff
@@ -1518,63 +1530,68 @@ func (dcr *ExchangeWallet) sendWithReturn(baseTx *wire.MsgTx, addr dcrutil.Addre
-			// And there is likely no room for improvement. If 2) The minFee
+
+			// If 1) lastFee == reqFee, nothing changed since the last cycle.
+			// And there is likely no room for improvement. If 2) The reqFee
 			// required for a transaction of this size is less than the
 			// currently signed transaction fees, but we've already tried it,
 			// then it must have a larger serialize size, so the current fee is
 			// as good as it gets.
-			if lastFee == minFee || (minFee < lastFee && tried[minFee] == 1) {
+			if lastFee == reqFee || (lastFee > reqFee && tried[reqFee]) {
 				break
 			}
+
 			// The minimum fee for a transaction of this size is either higher or
 			// lower than the fee in the currently signed transaction, and it hasn't
 			// been tried yet, so try it now.
-			tried[lastFee] = 1
-			subtractee.Value = int64(reservoir - minFee)
-			lastFee = minFee
+			tried[lastFee] = true
+			lastFee = reqFee
+			subtractee.Value = int64(reservoir - lastFee)
 			if dexdcr.IsDust(subtractee, feeRate) {
 				// Another condition that should be impossible, but check anyway in case
 				// the maximum fee was underestimated causing the first check to be
 				// missed.
-				dcr.log.Errorf("reached the impossible place. in = %d, out = %d, minFee = %d, lastFee = %d",
-					totalIn, totalOut, minFee, lastFee)
+				dcr.log.Errorf("reached the impossible place. in = %d, out = %d, reqFee = %d, lastFee = %d",
+					totalIn, totalOut, reqFee, lastFee)
 				return nil, nil, fmt.Errorf("dust error")
 			}
 			continue
 		}
 	}
-	// A couple of additional checks on the fee rate.
+
+	// Double check the resulting txns fee and fee rate.
 	checkFee, checkRate := fees(msgTx)
 	if checkFee != lastFee {
 		return nil, nil, fmt.Errorf("fee mismatch! %d != %d", checkFee, lastFee)
 	}
-	// If the integer truncated fee rate is not the same as the fee rate provided,
-	// log a warning.
-	if checkRate < float64(feeRate) {
-		return nil, nil, fmt.Errorf("final fee rate for %s, %f, is lower than expected, %d",
+	// Ensure the effective fee rate is at least the required fee rate.
+	if checkRate < feeRate {
+		return nil, nil, fmt.Errorf("final fee rate for %s, %d, is lower than expected, %d",
 			msgTx.CachedTxHash(), checkRate, feeRate)
 	}
 	// This is a last ditch effort to catch ridiculously high fees. Right now,
-	// it's just erroring for fees more than double the expected rate, which is
+	// it's just erroring for fees more than triple the expected rate, which is
 	// admittedly un-scientific. This should account for any signature length
 	// related variation as well as a potential dust change output with no
 	// subtractee specified, in which case the dust goes to the miner.
-	if checkRate > float64(feeRate*uint64(size)*2) {
-		return nil, nil, fmt.Errorf("final fee rate for %s, %f, is seemingly outrageous, %d",
+	if checkRate > feeRate*3 {
+		return nil, nil, fmt.Errorf("final fee rate for %s, %d, is seemingly outrageous, target = %d",
```

Comment:
No, but I think before it was comparing rate with total fee (was `checkRate > float64(feeRate*uint64(size)*2`, with the `size` multiplied there) so it's hard to know if double rate will sometimes be too strict especially if the rate is very low to begin with.  I was actually thinking about removing this though since btc doesn't do the same check and above it also verifies `checkFee != lastFee`.